tm/src/main/java/io/seata/tm/api/TransactionalTemplate.java

```diff
@@ -130,17 +131,17 @@ public boolean existingTransaction() {
 
     }
 
-
-
     private void completeTransactionAfterThrowing(TransactionInfo txInfo, GlobalTransaction tx, Throwable ex) throws TransactionalExecutor.ExecutionException {
         //roll back
         if (txInfo != null && txInfo.rollbackOn(ex)) {
             try {
-                rollbackTransaction(tx, ex);
+                if (tx.getRole() == GlobalTransactionRole.Launcher || txInfo.getParticipantReportRollback()) {
+                    rollbackTransaction(tx, ex);
+                }
```

Comment:
Changed the original logic, which will cause unParticipantReportRollback's participants not to take the TransactionHook.
Consider processing in rollbackTransaction-> tx.rollback.

Reply:
> Changed the original logic, which will cause unParticipantReportRollback's participants not to take the TransactionHook.
> Consider processing in rollbackTransaction-> tx.rollback.

changed